motionMonitor/motionMonitor.c

```diff
@@ -1,116 +1,158 @@
+    if (r != LE_OK)
+    { 
+        goto done;
+    }
+    *xAcc *= scaling;
 
-  pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompZ);
-  LE_ASSERT(pathLen < sizeof(path));
-  r = ioutil_readDoubleFromFile(path, zAcc);
-  *zAcc *= scaling;
+    pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompY);
+    LE_ASSERT(pathLen < sizeof(path));
+    r = ioutil_readDoubleFromFile(path, yAcc);
+    if (r != LE_OK)
+    {
+        goto done;
+    }
+    *yAcc *= scaling;
+
+    pathLen = snprintf(path, sizeof(path), FormatStr, AccType, CompZ);
+    LE_ASSERT(pathLen < sizeof(path));
+    r = ioutil_readDoubleFromFile(path, zAcc);
+    *zAcc *= scaling;
 
 done:
-  LE_INFO("Showing accel: x: %f, y: %f, z: %f", *xAcc, *yAcc, *zAcc);
-  return r;
+    LE_INFO("Showing accel X: %f Y: %f Z: %f ", *xAcc, *yAcc, *zAcc);
+    return r;
 }
 
-le_result_t brnkl_motion_getSuddenImpact(double* xAcc,
-                                         double* yAcc,
-                                         double* zAcc) {
-  *xAcc = suddenImpact.x;
-  *yAcc = suddenImpact.y;
-  *zAcc = suddenImpact.z;
-  suddenImpact.x = 0;
-  suddenImpact.y = 0;
-  suddenImpact.z = 0;
-  hasSuddenImpact = false;
-  return LE_OK;
+void recordImpact(double* xAcc, double* yAcc, double* zAcc){
+  timestamps[totalImpacts] = (unsigned long)time(NULL);
+  xAccImpact[totalImpacts] = *xAcc;
+  yAccImpact[totalImpacts] = *yAcc;
+  zAccImpact[totalImpacts] = *zAcc;
+  totalImpacts++;
+
+  LE_INFO("New Impact, totalImpacts: %d", totalImpacts);
 }
 
-int8_t brnkl_motion_hasSuddenImpact() {
-  return hasSuddenImpact;
+le_result_t brnkl_motion_getSuddenImpact(double* xAcc, size_t *xSize,
+              double* yAcc, size_t *ySize,
+              double* zAcc, size_t *zSize) {
+
+  if(!totalImpacts)
+    LE_INFO("No Sudden Impacts to Report");
+
+  *xSize = *ySize = *zSize = totalImpacts;
+
+  xAcc = xAccImpact;
+  yAcc = yAccImpact;
+  zAcc = zAccImpact;
+
+  totalImpacts = 0;
+
+  return LE_OK;
 }
 
-// take a reading but make sure its "larger" than
-// whatever is already in the buffer
-void interruptChangeHandler(bool state, void* ctx) {
+/*
+*Monitors accelerometer from iio on 100ms intervals
+*Sets hasSuddenImpact flag when accelerometer surpasses threshold
+*/
+void *impactMonitor(void * ptr){
   double x, y, z;
-  le_result_t r = brnkl_motion_getCurrentAcceleration(&x, &y, &z);
-  LE_INFO("TRIGGERED x:%f y:%f z:%f", x, y, z);
-  if (r == LE_OK) {
-    bool updateX = fabs(x) > fabs(suddenImpact.x);
-    bool updateY = fabs(y) > fabs(suddenImpact.y);
-    bool updateZ = fabs(z) > fabs(suddenImpact.z);
-    if (updateX) {
-      suddenImpact.x = x;
-    }
-    if (updateY) {
-      suddenImpact.y = y;
-    }
-    if (updateZ) {
-      suddenImpact.z = z;
-    }
-    if (updateX || updateY || updateZ) {
-      hasSuddenImpact = true;
-    }
+  for(;;){
+    brnkl_motion_getCurrentAcceleration(&x, &y, &z);
+
+    double euclidian = sqrt(x*x + y*y + z*z);
```

Comment:
Nitpick: I think it may be worth naming this variable after the quantity it represents.
```suggestion
    double impactMagnitude = sqrt(x*x + y*y + z*z);
```